File: common/dao-api/src/main/java/org/thingsboard/server/dao/testing/TestingService.java
/**
 * Copyright © 2016-2021 The Thingsboard Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.thingsboard.server.dao.testing;

import com.google.common.util.concurrent.ListenableFuture;
import org.thingsboard.server.common.data.*;
import org.thingsboard.server.common.data.device.DeviceSearchQuery;
import org.thingsboard.server.common.data.id.*;
import org.thingsboard.server.common.data.page.PageData;
import org.thingsboard.server.common.data.page.PageLink;
import org.thingsboard.server.dao.device.provision.ProvisionRequest;

import java.util.List;

public interface TestingService {

    TestingInfo findTestingInfoById(TenantId tenantId, TestingId deviceId);

    Testing saveTesting(Testing testing);

    PageData<Testing>findTestingsByTenantId(TenantId tenantId, PageLink pageLink);
    PageData<Testing> findTestingsByTenantIdAndType(TenantId tenantId, String sensorType, PageLink pageLink);
    Testing findTestingById(TenantId tenantId, TestingId testingId);
    void deleteTesting(TenantId tenantId, TestingId testingId);






}

File: ui-ngx/src/app/core/http/testing.service.ts
///
/// Copyright © 2016-2021 The Thingsboard Authors
///
/// Licensed under the Apache License, Version 2.0 (the "License");
/// you may not use this file except in compliance with the License.
/// You may obtain a copy of the License at
///
///     http://www.apache.org/licenses/LICENSE-2.0
///
/// Unless required by applicable law or agreed to in writing, software
/// distributed under the License is distributed on an "AS IS" BASIS,
/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
/// See the License for the specific language governing permissions and
/// limitations under the License.
///

import { Injectable } from '@angular/core';
import { defaultHttpOptionsFromConfig, RequestConfig } from './http-utils';
import { Observable, ReplaySubject } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { PageLink } from '@shared/models/page/page-link';
import { PageData } from '@shared/models/page/page-data';
import {
  ClaimRequest,
  ClaimResult,
  Testing,
  TestingCredentials,
  TestingInfo,
  DeviceSearchQuery
} from '@app/shared/models/testing.models';
import { EntitySubtype } from '@app/shared/models/entity-type.models';
import { AuthService } from '@core/auth/auth.service';
import { TestingRoutingModule } from '@app/modules/home/pages/testing/testing-routing.module';

@Injectable({
  providedIn: 'root'
})
export class TestingService {

  testingInfo:TestingInfo;
  constructor(
    private http: HttpClient
  ) {
   }
  

  //ktu beji ndryshimet

  public getTenantDeviceInfos(pageLink: PageLink, type: string = '',
                              config?: RequestConfig): Observable<PageData<TestingInfo>> {
    return this.http.get<PageData<TestingInfo>>(`/api/tenant/testings${pageLink.toQuery()}&type=${type}`,
      defaultHttpOptionsFromConfig(config));
  }

  public getTenantDeviceInfos_(pageLink: PageLink,
  config?: RequestConfig): Observable<PageData<TestingInfo>> {
return this.http.get<PageData<TestingInfo>>(`/api/tenant/deviceInfos${pageLink.toQuery()}`,
defaultHttpOptionsFromConfig(config));
}
  public getTesting(testingId: string, config?: RequestConfig): Observable<Testing> {
    return this.http.get<Testing>(`/api/testing/${testingId}`, defaultHttpOptionsFromConfig(config));
  }

  public getTestings(testingIds: Array<string>, config?: RequestConfig): Observable<Array<Testing>> {
    return this.http.get<Array<Testing>>(`/api/testingss?testingsIds=${testingIds.join(',')}`, defaultHttpOptionsFromConfig(config));
  }

  public getTestingInfo(testingId: string, config?: RequestConfig): Observable<TestingInfo> {
    return  this.http.get<TestingInfo>(`/api/testing/${testingId}`, defaultHttpOptionsFromConfig(config));

    //return this.http.get<TestingInfo>(`/api/testing/info/${testingId}`, defaultHttpOptionsFromConfig(config));
  }

  public saveTesting(testing: Testing, config?: RequestConfig): Observable<Testing> {
    return this.http.post<Testing>('/api/testing', testing, defaultHttpOptionsFromConfig(config));
  }

  public deleteDevice(deviceId: string, config?: RequestConfig) {
    return this.http.delete(`/api/testing/${deviceId}`, defaultHttpOptionsFromConfig(config));
  }

  public getDeviceTypes(config?: RequestConfig): Observable<Array<EntitySubtype>> {
    return this.http.get<Array<EntitySubtype>>('/api/device/types', defaultHttpOptionsFromConfig(config));
  }

}
